### PR TITLE
fix #3020 by enabling Assembly.EntryPoint on CoreCLR

### LINKED ISSUE #3020: Support Startup.Main entrypoint
Just in case we don't get this for RC https://github.com/dotnet/corefx/issues/3336

/cc @muratg

### PR DESCRIPTION
**Pull request by anurse**
Fixes #3020 

/cc @davidfowl

### CODE REVIEW

File: NuGet.config [hunk 1]
```diff
@@ -1,7 +1,8 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <configuration>
   <packageSources>
-    <add key="AspNetVNext" value="https://www.myget.org/F/aspnetcidev/api/v3/index.json" />
+    <add key="dotnet-core" value="https://www.myget.org/F/dotnet-core/api/v3/index.json" />
```
Comment (anurse) [inline NuGet.config]: I'll remove this before merge. Have it here so I can see if https://github.com/dotnet/corefx/pull/3999 gets to the build in time and replace the reflection with that.

**Comment by davidfowl** [comment]
:shipit:

**Comment by anurse** [comment]
Once `build verify` passes I'll remove `dotnet-core` from the NuGet.Config and merge. Private reflection is good enough for now.

File: NuGet.config [hunk 2]
```diff
@@ -1,7 +1,7 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <configuration>
   <packageSources>
-    <add key="AspNetVNext" value="https://www.myget.org/F/aspnetcidev/api/v3/index.json" />
```
Comment (moozzyk) [inline NuGet.config]: @anurse - Should this have been checked in?

Comment (anurse) [inline NuGet.config]: Ah shoot, no.

Comment (anurse) [inline NuGet.config]: Fixing now

**Pull request merged**